rpc/server.go

```diff
@@ -404,6 +397,8 @@ func NewServer(logger utils.ZapCompatibleLogger, opts ...ServerOption) (Server,
 		serverOpts = append(serverOpts, grpc.StatsHandler(sOpts.statsHandler))
 	}
 
+	serverOpts = append(serverOpts, grpc.WaitForHandlers(sOpts.waitForHandlers))
```

Comment:
@dgottlieb another option is to just have `grpcServer` call `GracefulStop()` instead of `Stop()` in [L864](https://github.com/viamrobotics/goutils/blob/d2f65e2ad6b070fb5395d35b7af7e1b727971632/rpc/server.go#L864). It is a riskier change but also not sure why that Stop() wasn't graceful, when all of the other stops in the function are.

Reply:
Yeah, `GracefulStop` works too. I have no strong opinions on this. Just confirming: I expect all servers to be constructed from this `NewServer` call. Given that, I don't see any behavior difference between using an option vs GracefulStop.